packages/typescript-eslint/src/config-helper.ts

```diff
@@ -94,6 +94,7 @@ export function config(
     const extension = {
```

Comment:
This will work of course to solve the issue, but wondering if this should be 
`Object.from(Object.entries(config).filter([key, value] => value != null))`
or somesuch instead in order to pass along any other unenumerated properties?

Reply:
Nah -- we're not trying to include _every_ property. This is here to declare a set of overrides that the "base" config applies to the extended config. All other properties come from the extended config.

For example:
```ts
const extended = {
  file: ['a'],
  ignores: ['b'],
  name: 'c',
  rules: { foo: 1 },
};
const base = config({
  extends: [extended],
  file: ['x'],
  ignores: ['y'],
  name: 'z',
  rules: { bar: 1 },
});

// result

const base = [
  {
    file: ['x'],
    ignores: ['y'],
    name: 'z',
    rules: { foo: 1 },
  },
  {
    file: ['x'],
    ignores: ['y'],
    name: 'z',
    rules: { bar: 1 },
  }
];
```

Anything not in this object will just derive from the extended config